### DISCUSSION TITLE
ipv6?

### DISCUSSION THREAD

**Discussion by Yet-another-NPC-guy** [StevenBlack/hosts#2718] [Q&A]
I'm using hosts on Android with AdAway root method(direct hosts file modification). On AdAway I can enable ipv6 :: and it will duplicate every domain on the hosts with the ip ::. My question is: It's necessary on Android and Unix based systems? My network is ipv4/ipv6.

@StevenBlack

**Comment by StevenBlack** [comment]
Thank you for your question @Hell0Earth.

Hosts files, like everything DNS-related, translates plaintext domain names into IP addresses so traffic from apps and browsers can be routed where it needs to go.

This IP address resolution is therefore mostly to direct traffic beyond your network.  

In the case of this project, the `0.0.0.0` address resolution helps ensure that certain requests never leave your machine.

So the hosts files we provide are to prevent traffic intended for specific domains from ever leaving your machine.  Therefore it doesn't really matter whether your network is IPv4 or IPv6.

Is this helpful? I'm happy to respond to clarifications you may need.

**Reply by Yet-another-NPC-guy** [reply]
Thanks for give me your time.

So if I understand on hosts file when the finality is block certain domains, is irrelevant if the assigned ip is ipv4: "0.0.0.0" or ipv6: "::" the system will use the first IP that matches with the domain name and due to both ip are for blocking, domain won't be loaded? 

If I'm correct why hosts support ipv6 and why chatgpt says that if I don't add all domains with both ip, ipv6 and ipv4 may some sistes can be loaded if my network is ipv6/ipv4? 

Sorry for my bad English and again, thanks for your time.

I asked to gpt because I can't found complete information about how hosts file works on Unix based systems and that is why I'm taking your time. Hosts, dns users community is big but the information isn't big 😔

**Reply by StevenBlack** [reply]
@Hell0Earth it's actually much simpler than you imagine.

Your machine understands IPv4 and IPv6.  The hosts file is always polled first. If the hosts file returns `0.0.0.0` (or the equivalent IPv6) for a particular domain, the request never leaves your machine.

All this happens way above the network layer. The request never leaves the application layer, and your network (several layers below) never enters.

**Reply by Yet-another-NPC-guy** [reply]
> @Hell0Earth it's actually much simpler than you imagine.
> 
> Your machine understands IPv4 and IPv6.  The hosts file is always polled first. If the hosts file returns `0.0.0.0` (or the equivalent IPv6) for a particular domain, the request never leaves your machine.
> 
> All this happens way above the network layer. The request never leaves the application layer, and your network (several layers below) never enters.

Thanks so much for your help.
Now I understand, in conclusion with ipv4 0.0.0.0 is sufficient, ipv6 it's not necessary because hosts doesn't works at network level so for the hosts is irrelevant if the network is ipv4 or ipv6, hosts will assign the first IP that matches the domain name. Well I will disable ipv6, now I have around 400.000 domains due to duplications with ipv6 :: Unnecessary waste resources XD. Thanks so much for the help and your time, have a good afternoon ✌️
